### PR TITLE
Account for stream as a satellite release version

### PR DESCRIPTION
**Pull request by JacobCallahan**
Since we are setting stream as the satellite release version, we need to fallback to the backup satellite version in config.

### CODE REVIEW

File: robottelo/hosts.py [hunk 1]
```diff
@@ -71,11 +71,9 @@ def get_sat_version():
...
-        if hasattr(settings.server.version, 'release'):
-            sat_version = str(settings.server.version.release)
-        elif hasattr(settings.robottelo, 'satellite_version'):
-            sat_version = settings.robottelo.satellite_version
-        else:
+        if str(sat_version := settings.server.version.get('release')) == 'stream':
+            sat_version = str(settings.robottelo.get('satellite_version'))
+        if not sat_version:
             sat_version = SATELLITE_VERSION
     return Version('9999' if 'nightly' in sat_version else sat_version)
```
Comment (devendra104) [inline robottelo/hosts.py:78]: `stream` string doesn't comply with the PEP-440 (https://github.com/pypa/packaging/blob/5b34465682d40ddd42dd6cba88e158e1674de7b9/src/packaging/version.py#L113)

**Review by jyejare — COMMENTED** [review]
ACK pending questions !

File: robottelo/hosts.py (hunk 1 above)
Comment (JacobCallahan) [reply robottelo/hosts.py:78]: correct. the fix here is to use settings.robottelo.satellite_version if settings.server.version.release is set to stream.

File: robottelo/hosts.py [hunk 2]
```diff
@@ -71,11 +71,9 @@ def get_sat_version():
     try:
         sat_version = Satellite().version
     except (AuthenticationError, ContentHostError, BoxKeyError):
-        if hasattr(settings.server.version, 'release'):
-            sat_version = str(settings.server.version.release)
-        elif hasattr(settings.robottelo, 'satellite_version'):
-            sat_version = settings.robottelo.satellite_version
-        else:
+        if str(sat_version := settings.server.version.get('release')) == 'stream':
+            sat_version = str(settings.robottelo.get('satellite_version'))
```
[the remark this answers is not in the document]
Comment (JacobCallahan) [reply robottelo/hosts.py:75]: correct. we should always be setting settings.robottelo.satellite_version so it can be the fallback here

**Review by devendra104 — APPROVED** [review]
ACK

**Comment by ogajduse** [comment]
This PR should have been cherry-picked to `z` branches too. Now it is causing cherry-pick process to fail.

**Comment by JacobCallahan** [comment]
The manifester version could have come back. However, the other change didn't need to.

**Pull request merged**